src/main/java/com/google/step/datamanager/DealManagerDatastore.java

```diff
@@ -141,21 +145,42 @@ public Deal updateDeal(Deal deal, List<String> tagNames) {
     return transformEntityToDeal(dealEntity);
   }
 
+  /** Retrieves deals posted by restaurants or users */
+  private List<Deal> getDealsPublishedByRestaurantsOrUsers(
+      Set<Long> idsOfFollowedFieldName, String filterAttribute) {
+    List<Deal> dealResults = new ArrayList<>();
+    for (Long id : idsOfFollowedFieldName) {
+      Filter propertyFilter = new FilterPredicate(filterAttribute, FilterOperator.EQUAL, id);
+      Query query = new Query("Deal").setFilter(propertyFilter);
+      PreparedQuery pq = datastore.prepare(query);
+      for (Entity entity : pq.asIterable()) {
+        dealResults.add(readDeal(entity.getKey().getId()));
+      }
+    }
```

Comment:
If you use `FilterOperator.IN` instead of `FilterOperator.EQUAL`, can you make this a single call to datastore?